src/building/compiler-documenting.md

```diff
@@ -1,55 +1,50 @@
 # Building documentation
 
-You might want to build documentation of the various components
-available like the standard library. There’s two ways to go about this.
-You can run rustdoc directly on the file to make sure the HTML is
-correct, which is fast. Alternatively, you can build the documentation
-as part of the build process through `x.py`. Both are viable methods
-since documentation is more about the content.
+This chapter describes how to build documentation of toolchain components,
+either in whole or individually.
 
-## Document everything
+- Document everything
 
-This uses the beta rustdoc, which usually but not always has the same output
-as stage 1 rustdoc.
+  This uses `rustdoc` from the beta toolchain,
+  so will produce (slightly) different output to stage 1 rustdoc,
+  as `rustdoc` is under active development:
 
-```bash
-./x.py doc
-```
+  ```bash
+  ./x.py doc
+  ```
 
-## If you want to be sure that the links behave the same as on CI
+  If you want to be sure the documentation looks the same as on CI:
 
-```bash
-./x.py doc --stage 1
-```
+  ```bash
+  ./x.py doc --stage 1
+  ```
 
-First the compiler and rustdoc get built to make sure everything is okay
-and then it documents the files.
+  First,
+  the compiler and rustdoc get built to make sure everything is okay,
+  and then it documents the files.
 
-## Document specific components
+- Much like running individual tests or building specific components,
+  you can build just the documentation you want:
 
-```bash
-./x.py doc src/doc/book
-./x.py doc src/doc/nomicon
-./x.py doc src/doc/book library
-```
+  ```bash
+  ./x.py doc src/doc/book
+  ./x.py doc src/doc/nomicon
+  ./x.py doc compiler library
+  ```
 
-Much like individual tests or building certain components you can build only
- the documentation you want.
+- Document internal rustc items
 
-## Document internal rustc items
+  Compiler documentation is not built by default.
+  To enable it, modify `config.toml`:
```

Comment:
This was changed a while back to only control the defaults, not be a hard-off switch. https://github.com/rust-lang/rust/pull/90683
```suggestion
  To create it by default with `x.py doc`, modify `config.toml`:
```

Reply:
What does this mean... what's the difference?

Reply:
Before if you ran `x.py doc compiler ` it would say "skipping because compiler-docs is disabled". Now it builds the docs in that case, but not if you only run `x.py doc`.